client/lib/plans/index.js

```diff
@@ -66,6 +68,28 @@ export function getSitePlanSlug( siteID ) {
 	return get( site, 'plan.product_slug' );
 }
 
+function canUpgradeToPlan( planKey, site ) {
+	const offer = site.jetpack ? JETPACK : WPCOM;
```

Comment:
Also, I would use `isJetpackSite` selector from `state/sites/selectors` to avoid DRY and keep things consistent.